universe/Universe.cpp

```diff
@@ -1660,6 +1660,21 @@ void Universe::ForgetKnownObject(int empire_id, int object_id) {
     objects.Remove(object_id);
 }
 
+void Universe::CheckShipDesignVisibility(int empire_id, int object_id, Visibility vis) {
```

Comment:
No, I certainly don't think it's the same.  I thought you were objecting to SetShipDesignVisibility because we are not really setting visibility like we do for objects.  This is the function that is responsible for granting knowledge of ship designs, and it seems to me that GrantShipDesignKnowledge is a very good name for it; that is its essence.

In trying to guess why you think my proposed names would be misleading and yours not, I can only see that your function name mentions the requirement of the ship being visible.  Now, if we have one function that handled this for when the ships are visible, and a different function that granted ship design knowledge in other situations, then I could see that the distinction should be part of the name.  But I don't see that in this case.

If the premise is that the function name needs to include all relevant conditions for operation regardless of whether they are needed to distinguish from some other function, then it seems we should really be requiring the name to mention that it is Partial visibility being required not just Basic.  But, mentioning the ship visibility in the name at all, let alone with the "Partial" reference, just seems pointlessly long to me.  

But perhaps I am misunderstanding your reasoning?